### ISSUE TITLE
filter() does not work after 1.9.0 if noFill() is called

### ISSUE THREAD

**Issue by inaridarkfox4231** [processing/p5.js#7425]
### Topic

filter() was introduced in webgl mode in 1.8.0. At this time, it seems that filter() was still working even with noFill().

```js
// version: 1.8.0
function setup() {
   createCanvas(200,200,WEBGL);
 }
  
 function draw() {
   background(0);
   noFill();
   stroke(255);
   line(-100,-100,300,300);
   filter(INVERT);
 }
```

![cwcdcwvr3v3](https://github.com/user-attachments/assets/7fcc7259-a740-48de-9a2c-3c34b23bad99)

However, since 1.9.0, filter() does not work if noFill() is called. Of course it doesn't work with 1.11.2 either.

![r3f3f33f33f](https://github.com/user-attachments/assets/4d88e8ec-941f-4217-8a67-681d452814fd)

It will be drawn when you execute fill().
```js
function setup() {
    createCanvas(200,200,WEBGL);
}

function draw() {
    background(0);
    noFill();
    stroke(255);
    line(-100,-100,300,300);
    
    fill(255);
    
    filter(INVERT);
}
```

![ef3d3frf3f3f3](https://github.com/user-attachments/assets/4228f44a-de34-48a6-92f2-fbb262f5ad66)

It seems nonsensical to have to call pointless fill() to filter a sketch that just draws lines.
But I don't know if this is a bug or not.

That's all for raising the issue.

**Comment by inaridarkfox4231** [comment]
here: [noFill_filter_BUG](https://editor.p5js.org/dark_fox/sketches/OrovB6ZMj)

## noFill() and no "fill()"
```js
function setup() {
  createCanvas(400, 400, WEBGL);

  rotateX(PI/4);
  rotateY(PI/4);
  noFill();
  stroke(255);
  background(0);
  box(200);

  filter(INVERT);
}
```

![hjhgwhdghvfbh](https://github.com/user-attachments/assets/be8be08f-8478-45fa-b563-505171ab195b)

## noFill() and "fill()"

```js
function setup() {
  createCanvas(400, 400, WEBGL);

  rotateX(PI/4);
  rotateY(PI/4);
  noFill();
  stroke(255);
  background(0);
  box(200);

  fill("aquamarine");

  filter(INVERT);
}
```

![efefefrf3ffr4f4](https://github.com/user-attachments/assets/085efb3b-d700-4d29-8370-745d7bc9e5b0)

**Comment by subhraneel2005** [comment]
yes, if nofill() is given then giving fill() is uneccesary, thanks for noticing it. will work on this

**Comment by inaridarkfox4231** [comment]
Thank you for your working on this issue. I'm not familiar with the behavior around filter() and framebuffer, so I can't work for this issue.

**Comment by subhraneel2005** [comment]
its all good, no worries 😊

**Comment by subhraneel2005** [comment]
i think the bug was with noFill().
just added `this._renderer.fill(255);` to it. So, it will have a white fill by default, and we dont have to call fill() again. what do u think ?
![sol1](https://github.com/user-attachments/assets/a7c436b9-fd1a-422d-a6c7-8bd634717ba8)

**Comment by davepagurek** [comment]
Just a heads up -- In the `dev-2.0` branch, where p5 2.0 is being developed, this section of the code has been refactored, so the implementation will likely be a little different. We now store the fill color as `_renderer.states.fillColor`, which is null if there should not be a fill, and is a `p5.Color` if a fill is active.

**Comment by Vaivaswat2244** [comment]
Is someone working on this? If not I'd like to proceed on this.
Also instead of providing the `fillColor` with null. I'm assuming making it transparent white would be a better fix,
something on the lines of,
```javascript
p5.prototype.noFill = function() {  
  this._renderer.states.fillColor = this.color(255, 255, 255, 0); // Transparent white
  return this;  
};
```

**Comment by davepagurek** [comment]
There's a performance (and depth ordering) implication to having fill set, since we have to triangulate all the faces in order to fill them, and then it has an entry in the depth buffer, preventing farther-away objects from being drawn at those pixels. So I think we still want to avoid doing fills at all if `noFill` has been set, but ideally just update the `filter()` logic to run regardless of the fill.
